R/adducts.R

```diff
@@ -0,0 +1,449 @@
+#' @title Calculate mass-to-charge ratio
+#'
+#' @description
+#'
+#' `mass2mz` calculates the m/z value from a neutral mass and an adduct 
+#'      definition.
+#'
+#' @param x `numeric` neutral mass for which the adduct m/z shall be calculated.
+#'
+#' @param adduct `character` Valid adduct definition.
+#'
+#' @return `numeric` m/z value calculated
+#'
+#' @author Michael Witting
+#'
+#' @export
+#'
+#' @examples
+#'
+#' exact_mass <- 100
+#' adduct <- "[M+H]+"
+#' 
+#' ## Calculate m/z of [M+H]+ adduct from neutral mass
+#' mass2mz(exact_mass, adduct)
+#' 
+#' exact_mass <- 100
+#' adduct <- "[M+Na]+"
+#' 
+#' ## Calculate m/z of [M+Na]+ adduct from neutral mass
+#' mass2mz(exact_mass, adduct)
+mass2mz <- function(x, adduct = "[M+H]+") {
+  
+  adduct_rules_all <- c(.adductRules(polarity = "positive"),
+                        .adductRules(polarity = "negative"))
+
+  # check if adduct suppplied is in the list of valid adducts
+  if(!adduct %in% names(adduct_rules_all)) {
+    
+    stop(paste0("Unknown adduct: ", adduct))
+    
+  }
+  
+  # retrieve multiplicative and additive part for calculation
+  mass_multi <- adduct_rules_all[[adduct]]$mass_multi
+  mass_add <- adduct_rules_all[[adduct]]$mass_add
+  
+  ion_mass <- x * mass_multi + mass_add
+  
+  ion_mass
+  
+}
+
+#' @title Calculate neutral mass
+#'
+#' @description
+#'
+#' `mz2mass` calculates the neutral mass from a given m/z value and adduct 
+#'      definition
+#'
+#' @param x `numeric` m/z value for which the neutral mass shall be calculated.
+#'
+#' @param adduct `character` Valid adduct definition.
```

Comment:
You could use here a `@inheritParams mass2mz` and drop the definition of `@param adduct` - roxygen will then use the definition of this parameter in `mass2mz`.